src/crab_utils/graph_ops.hpp

```diff
@@ -546,16 +553,17 @@ class GraphOps {
         while (new_sz < sz)
             new_sz = static_cast<size_t>(new_sz * 1.5);
 
-        check_realloc((void**)&edge_marks, sizeof(char) * new_sz * new_sz);
-        check_realloc((void**)&dual_queue, sizeof(vert_id) * 2 * new_sz);
-        check_realloc((void**)&vert_marks, sizeof(int) * new_sz);
+        edge_marks->resize(new_sz * new_sz);
+        dual_queue->resize(2 * new_sz);
+        vert_marks->resize(new_sz);
```

Comment:
This change (moving from a pointer to vector and from realloc to vector resize) is significant and not technically part of the PR. It might be a good idea to have it on a dedicated commit.

(I'm pretty sure I tried it at some point. I don't remember what the issue was. Looks like it's fine now.)

Reply:
My reason for including it is that it significantly simplifies the cleanup of memory and allows the memory used by this to be tracked via an instrumented global new operator.